### PR TITLE
Introduce Logging

### PR DESCRIPTION
**Pull request by williamcroberts**
I wanted to introduce a logging library that can be coupled into moving the tools over. For verbose info, capturing -v and calling log_set_level() would be sufficient for getting the debug information. This sets us up for killing off debug.c and debug.h.

From here we can start fixing up the other things to start using LOG_(INFO|WARN|ERR) based on need.

### CODE REVIEW

File: src/tpm2_send_command.c [hunk 1]
```diff
@@ -37,110 +37,96 @@
...
                  file_stream);
     if (ret != 1 && ferror (file_stream)) {
-        fprintf (stderr,
-                 "Failed to read command body: %s\n",
+        LOGE ("Failed to read command body: %s\n",
                  strerror (errno));
         return 0;
     }
-    LOG_VERBOSE (stderr, "  read command body successfully\n");
+    LOGV ("read command body successfully\n");
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 2]
```diff
@@ -37,110 +37,96 @@
...
+    LOGV ("command size: 0x%08x", get_command_size (buf));
+    LOGV ("command code: 0x%08x", get_command_code (buf));
     ret = fread (buf + COMMAND_HEADER_SIZE,
                  command_size - COMMAND_HEADER_SIZE,
                  1,
                  file_stream);
     if (ret != 1 && ferror (file_stream)) {
-        fprintf (stderr,
-                 "Failed to read command body: %s\n",
+        LOGE ("Failed to read command body: %s\n",
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 3]
```diff
@@ -37,110 +37,96 @@
...
+    LOGV ("write_response_to_file");
     num_write = fwrite (buf, RESPONSE_HEADER_SIZE, 1, file_stream);
     if (num_write != 1 && ferror (stdout)) {
-        fprintf (stderr, "failed to write to stdout\n");
+        LOGE ("failed to write to stdout");
         return 0;
     }
-    LOG_VERBOSE (stderr,
-                 "  wrote response header: %ld members of size 0x%lx\n",
+    LOGV ("wrote response header: %ld members of size 0x%lx\n",
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 4]
```diff
@@ -37,110 +37,96 @@
...
                  num_write, RESPONSE_HEADER_SIZE);
     response_size = get_command_size (buf);
     if (response_size > buf_size) {
-        fprintf (stderr,
-                 "buf_size is insufficient: %u > %ld\n",
+        LOGE ("buf_size is insufficient: %u > %ld\n",
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 5]
```diff
@@ -37,110 +37,96 @@
...
-        fprintf (stderr,
-                 "Failed to write response to file: %s\n",
+        LOGE ("Failed to write response to file: %s",
                  strerror (errno));
         return 0;
 
     }
-    LOG_VERBOSE (stderr,
-                 "  wrote response body: %zd members of size 0x%x\n",
+    LOGV ("wrote response body: %zd members of size 0x%x\n",
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 6]
```diff
@@ -164,37 +150,34 @@ execute_tool (int              argc,
     UINT32 written_size, command_size;
     TSS2_RC rc;
 
-    verbose = opts->verbose;
-
     command_size = read_command_from_file (stdin, buf, MAX_BUF);
     if (command_size == 0) {
-        fprintf (stderr, "failed to read TPM2 command buffer from stdin\n");
+        LOGE ("failed to read TPM2 command buffer from stdin\n");
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 7]
```diff
@@ -164,37 +150,34 @@ execute_tool (int              argc,
...
         return 1;
     }
 
     rc = Tss2_Sys_GetTctiContext (sapi_context, &tcti_context);
     if (rc != TSS2_RC_SUCCESS) {
-        fprintf (stderr,
-                 "Failed to get TCTI context from SAPI context: 0x%x\n",
+        LOGE ("Failed to get TCTI context from SAPI context: 0x%x\n",
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 8]
```diff
@@ -164,37 +150,34 @@ execute_tool (int              argc,
...
                  rc);
         return 1;
     }
     rc = tss2_tcti_transmit (tcti_context, command_size, buf);
     if (rc != TSS2_RC_SUCCESS) {
-        fprintf (stderr, "tss2_tcti_transmit failed: 0x%x\n", rc);
+        LOGE ("tss2_tcti_transmit failed: 0x%x\n", rc);
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 9]
```diff
@@ -164,37 +150,34 @@ execute_tool (int              argc,
...
         return 1;
     }
     rc = tss2_tcti_receive (tcti_context,
                             &response_size,
                             buf,
                             TSS2_TCTI_TIMEOUT_BLOCK);
     if (rc != TSS2_RC_SUCCESS) {
-        fprintf (stderr, "tss2_tcti_receive failed: 0x%x\n", rc);
+        LOGE ("tss2_tcti_receive failed: 0x%x\n", rc);
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

File: src/tpm2_send_command.c [hunk 10]
```diff
@@ -164,37 +150,34 @@ execute_tool (int              argc,
...
         return 1;
     }
     written_size = write_response_to_file (stdout, buf, response_size);
     if (written_size < response_size) {
-        fprintf (stderr, "write_command_to_file failed\n");
+        LOGE ("write_command_to_file failed\n");
```
Comment (williamcroberts) [inline src/tpm2_send_command.c]: new line needs to be removed

**Comment by flihp** [comment]
merged

**Pull request closed without merging**